src/components/ProjectDetail/TaskFilter/useGetTaskListFilter.ts

```diff
@@ -0,0 +1,54 @@
+import { getUser } from 'utils/auth';
+import { nonAccentVietnameses } from 'utils/convert';
+import { Task, TaskStatus } from 'utils/types';
+import { TasksFilter } from '../ProjectTasks';
+
+export const useGetTaskListByFilter = (filter: TasksFilter, tasks: Task[]) => {
+   const currentUser = getUser();
+   const filterKeys = Object.keys(filter);
+   let tasksFilter = tasks.filter((task) => {
+      return filterKeys.every((eachKey) => {
+         if (
+            ((filter[eachKey] && !filter[eachKey].length) ||
+               filter[eachKey] === null) &&
+            typeof filter[eachKey] !== 'boolean'
+         ) {
+            return true;
+         }
+         if (filter[eachKey].length === 0 && Array.isArray(filter[eachKey])) {
+            return true;
+         }
+
+         switch (eachKey) {
+            case 'search':
+               const searchWord: string = nonAccentVietnameses(filter[eachKey]);
+               return (
+                  nonAccentVietnameses(task.title).includes(
+                     nonAccentVietnameses(searchWord)
+                  ) ||
+                  nonAccentVietnameses(task.notes!).includes(
+                     nonAccentVietnameses(searchWord)
+                  )
+               );
+            case 'createBy':
+               return (
+                  (task.requestByUser.id === currentUser?.id) ===
+                  filter[eachKey]
+               );
+            case 'assignTo':
+               return (task.assign?.id === currentUser?.id) === filter[eachKey];
+            case 'statuses':
+               const _filter = Array.from(filter[eachKey], (item) =>
+                  TaskStatus[item].toString()
+               );
+
+               return _filter.includes(task.status.toString());
```

Comment:
```suggestion
               return filter['statuses'].includes(task.status)
```


Không cần thiết phải filter lằng nhằng quá đâu e, filter['statuses'] và task.status có cùng kiểu rồi, e mất thêm 1 bước để convert nữa